### PR TITLE
Use new S3 naming scheme and add optional sha parameter.

### PR DESCRIPTION
**Pull request by mberhault**
* optionally specify a build sha for all binaries
* lookup latest sha in scripts and pass it to terraform (avoids the ssh)
* add link to sha (rely on gmail linkifying messages)

### CODE REVIEW

File: scripts/utils.sh [hunk 1]
```diff
@@ -14,3 +14,46 @@ do_retry() {
     fi
   done
 }
+
+# latest_sha takes a [repo]/[binary] and looks up the latest  sha in S3.
```
Comment (tschottdorf) [inline scripts/utils.sh]: double space

**Comment by tschottdorf** [comment]
LGTM. Just to double-check, we're getting rid of all of those scripts, and shouldn't add anything in there that's not immediately necessary, correct?

File: scripts/utils.sh (hunk 1 above)
Comment (mberhault) [inline scripts/utils.sh]: done

**Comment by mberhault** [comment]
this was mostly about using the naming scheme so that I could have the sha there and do proper GC on the S3 bucket. 
but yes, my next step will be to move more of this stuff to go. Removing the extra ssh command to get the sha will help towards that.

**Pull request merged**